app/Http/Controllers/TalksController.php

```diff
@@ -141,72 +142,70 @@ public function show($id)
 
     public function destroy($id)
     {
-        Auth::user()->talks()->withTrashed()->findOrFail($id)->forceDelete();
+        Auth::user()->talks()->findOrFail($id)->delete();
 
         Session::flash('message', 'Successfully deleted talk.');
 
         return Redirect::to('talks');
     }
 
-    public function softDelete($id)
+    public function archiveIndex()
     {
-        Auth::user()->talks()->findOrFail($id)->delete();
+        $talks =  Auth::user()->talks()->archived()->get();
+        list($talks, $sorting_talk) = $this->sortTalks($talks);
+
+        return View::make('talks.archive')
+          ->with('talks', $talks)
+          ->with('sorting_talk', $sorting_talk);
+    }
+
+    public function archive($id)
+    {
+        $talk = Auth::user()->talks()->findOrFail($id);
+        $talk->is_archived = true;
+        $talk->save();
 
         Session::flash('message', 'Successfully archived talk.');
 
         return Redirect::to('talks');
     }
 
-    public function restore($id)
+    public function restore($talkId)
     {
-        Auth::user()->talks()->withTrashed()->findOrFail($id)->restore();
+        $talk = Auth::user()->talks()->findOrFail($talkId);
+        $talk->is_archived = false;
+        $talk->save();
 
         Session::flash('message', 'Successfully restored talk.');
 
         return Redirect::to('archive');
     }
 
-
-    public function archiveIndex()
-    {
-        list($talks, $sorting_talk) = $this->sortTalks('archive');
-
-        return View::make('talks.archive')
-          ->with('talks', $talks)
-          ->with('sorting_talk', $sorting_talk);
-    }
-
   /**
    * Sorts through talks and returns an array of variables. Access the variables using the list() construct.
    * @param  string $type Whether the index is for archived talks or active talks
    * @return array       Returns array of variables.
    */
-    private function sortTalks($type = 'active')
+    private function sortTalks($talks)
     {
         $bold_style = 'style="font-weight: bold;"';
```

Comment:
Please get rid of `$Bold_style`. Instead, if a sort is chosen, set it to be `true` instead of `false`; then in the template, echo this same style *if* the given sort is `true`